### PR TITLE
committing changes for DE2440.

### PR DESCRIPTION
**Pull request by ainderkkumar**
Requesting these changes to be merged . We need to add the CA cert contents to be added to the TrustManager so that we can establish a HTTPS connection to Swarm Manager VM from Agility Platform using Swarm Service Adapter.

### CODE REVIEW

**Comment by servicemesh-autobuild** [comment]
:white_check_mark: Build was successful :tada:


http://ci.servicemesh.com:8080/job/csc-agility-platform-sdk-prb/65/

File: agility.io/src/main/java/com/servicemesh/io/http/impl/DefaultHttpClient.java [hunk 1]
```diff
@@ -380,14 +380,23 @@ else if (requestTimeout != null)
 
     private void setSSLContext(final HttpAsyncClientBuilder builder, final IHttpClientConfig config)
     {
-        TrustManager trustManager = new EasyTrustManager();
+        TrustManager[] trustManagers = null;
+        if (config != null && config.getTrustManagers() != null)
```
Comment (rafsanchez) [inline agility.io/src/main/java/com/servicemesh/io/http/impl/DefaultHttpClient.java]: This should also test that config.getTrustManagers().length > 0

File: agility.io/src/main/java/com/servicemesh/io/http/impl/DefaultHttpClient.java [hunk 2]
```diff
@@ -380,14 +380,23 @@ else if (requestTimeout != null)
...
+            trustManagers = new TrustManager[] { trustManager };
+        }
         KeyManager[] keyManagers = (config != null) ? config.getKeyManagers() : null;
 
         try
         {
             SSLContext sslContext = SSLContext.getInstance("TLS");
-
-            sslContext.init(keyManagers, new TrustManager[] { trustManager }, null);
+            //sslContext.init(keyManagers, new TrustManager[] { trustManager }, null);
```
Comment (rafsanchez) [inline agility.io/src/main/java/com/servicemesh/io/http/impl/DefaultHttpClient.java]: Remove this line.

File: agility.io/src/main/java/com/servicemesh/io/http/impl/DefaultHttpClient.java [hunk 3]
```diff
@@ -399,14 +408,25 @@ private void setSSLContext(final HttpAsyncClientBuilder builder, final IHttpClie
     private Registry<SchemeIOSessionStrategy> generateRegistry(final IHttpClientConfig config)
     {
         RegistryBuilder<SchemeIOSessionStrategy> builder = RegistryBuilder.create();
-        TrustManager trustManager = new EasyTrustManager();
+
+        TrustManager[] trustManagers = null;
+        if (config != null && config.getTrustManagers() != null)
```
Comment (rafsanchez) [inline agility.io/src/main/java/com/servicemesh/io/http/impl/DefaultHttpClient.java]: This should also test that config.getTrustManagers().length > 0

**Comment by rafsanchez** [comment]
Also a unit test that sets the trust manager with an empty array.

**Comment by servicemesh-autobuild** [comment]
:x: Build failed

http://ci.servicemesh.com:8080/job/csc-agility-platform-sdk-prb/66/

**Comment by servicemesh-autobuild** [comment]
:white_check_mark: Build was successful :tada:


http://ci.servicemesh.com:8080/job/csc-agility-platform-sdk-prb/67/

**Comment by servicemesh-autobuild** [comment]
:white_check_mark: Build was successful :tada:


http://ci.servicemesh.com:8080/job/csc-agility-platform-sdk-prb/68/

File: agility.io/src/test/java/com/servicemesh/io/http/HttpClientConfigTest.java [hunk 4]
```diff
@@ -69,9 +188,37 @@ public void testHttpClientConfig() throws Exception
...
         Assert.assertEquals(1, config.getKeyManagers().length);
+        Assert.assertNotNull(config.getTrustManagers());
+        Assert.assertEquals(1, config.getTrustManagers().length);
         Assert.assertNotNull(config.getCredentials());
         Assert.assertEquals(Credentials.CredentialsType.CREDENTIALS_TYPE_USERNAMEPASSORD, config.getCredentials().getType());
         Assert.assertNotNull(config.getProxy());
         Assert.assertEquals("localhost", config.getProxy().getHostname());
     }
+
+    public TrustManager[] getValidTrustManager() throws Exception
```
Comment (rafsanchez) [inline agility.io/src/test/java/com/servicemesh/io/http/HttpClientConfigTest.java]: Ok for now but this should really be private.

File: agility.io/src/test/java/com/servicemesh/io/http/HttpClientConfigTest.java [hunk 5]
```diff
@@ -69,9 +188,37 @@ public void testHttpClientConfig() throws Exception
...
+        KeyStore trustStore = createTrustStore(cacert);
+        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        trustManagerFactory.init(trustStore);
+        TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
+        return trustManagers;
+    }
+
+    public static KeyStore createTrustStore(String capem) throws Exception
```
Comment (rafsanchez) [inline agility.io/src/test/java/com/servicemesh/io/http/HttpClientConfigTest.java]: Ok for now but this should really be private.

**Comment by Behnke19** [comment]
Please make sure this gets merged to the develop branch as well. There is no automerge on this repo.
